### PR TITLE
Fix a few ADT library issues based on feedback so far

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/BasicRelationship.java [hunk 1]
```diff
@@ -118,13 +118,13 @@ public BasicRelationship setName(String name) {
...
 
     /**
-     * Sets the additional properties defined in the model. This field will contain any properties of the relationship that are not already defined by the other strong types of this class.
+     * Adds an additional property to this model. This field will contain any properties of the relationship that are not already defined by the other strong types of this class.
      * @param key The key of the additional property to be added to the relationship.
      * @param value The value of the additional property to be added to the relationship.
      * @return The BasicRelationship object itself.
      */
     @JsonAnySetter
-    public BasicRelationship setCustomProperties(String key, Object value) {
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/BasicRelationship.java:127]: A setter would take the full map, not a single key value pair. So I just renamed this to more accurately describe what this method does

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
 /**
  * Configures the {@link StdSerializer<String>} with custom behavior needed to work with Digital Twins REST API.
  */
-public class DigitalTwinsStringSerializer extends StdSerializer<String> {
+public final class DigitalTwinsStringSerializer extends StdSerializer<String> {
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java:16]: We don't plan on extending this class, so let's make it final

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java [hunk 3]
```diff
@@ -35,27 +35,6 @@
         this.digitalTwinsAsyncClient = digitalTwinsAsyncClient;
     }
 
-    /**
-     * Gets the {@link HttpPipeline} that this client is configured to use for all service requests. This pipeline can
-     * be customized while building this client through {@link DigitalTwinsClientBuilder#httpPipeline(HttpPipeline)}.
-     *
-     * @return The {@link HttpPipeline} that this client uses for all service requests.
-     */
-    public HttpPipeline getHttpPipeline() {
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:44]: This functions aren't present in the AppConfiguration sample project, and we don't need them

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java [hunk 4]
```diff
@@ -43,15 +43,14 @@
...
+@ServiceClient(builder = DigitalTwinsClientBuilder.class, isAsync = true)
 public final class DigitalTwinsAsyncClient {
     private static final ClientLogger logger = new ClientLogger(DigitalTwinsAsyncClient.class);
     private static final ObjectMapper mapper = new ObjectMapper();
-    private final DigitalTwinsServiceVersion serviceVersion;
     private final AzureDigitalTwinsAPIImpl protocolLayer;
     private static final Boolean includeModelDefinitionOnGet = true;
 
-    DigitalTwinsAsyncClient(HttpPipeline pipeline, DigitalTwinsServiceVersion serviceVersion, String host) {
+    DigitalTwinsAsyncClient(HttpPipeline pipeline, String host) {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:53]: How would we configure the service version?

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java (hunk 4 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:53]: It's already baked into the http pipeline instance here

Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:53]: Wait, no it isn't.

**Review by azabbasi — APPROVED** [review]
The only question is the service version and how it can be passed down to the client

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java (hunk 4 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java]: Even before this PR, we never actually used the service API version that was set in the client builder. Right now, our protocol layer has it hardcoded and has no way to set it. I'll touch base with Srikanta about this

**Pull request merged**